File: Bars.GisIntegration.Base/Entities/GisRole/GisRoleMethod.cs
namespace Bars.GisIntegration.Base.Entities.GisRole
{
    using Bars.B4.DataAccess;

    /// <summary>
    /// Метод, доступный для роли ГИС
    /// </summary>
    public class GisRoleMethod : BaseEntity
    {
        /// <summary>
        /// Роль ГИС
        /// </summary>
        public virtual GisRole Role { get; set; }

        /// <summary>
        /// Наименование доступного метода
        /// </summary>
        public virtual string MethodName { get; set; }

        /// <summary>
        /// exporter.GetType().Name доступного метода
        /// </summary>
        public virtual string MethodId { get; set; }
    }
}

File: Bars.GisIntegration.UI/libs/B4/model/gisrole/GisRoleMethod.js
Ext.define('B4.model.gisrole.GisRoleMethod', {
    extend: 'B4.base.Model',
    idProperty: 'Id',
    proxy: {
        type: 'b4proxy',
        controllerName: 'GisRoleMethod'
    },
    fields: [
        { name: 'Id', useNull: true },
        { name: 'Role', defaultValue: null },
        { name: 'MethodName' },
        { name: 'MethodId' }
    ]
});
